agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/localstorage/LocalFileLoader.java

```diff
@@ -140,28 +149,23 @@ PersistedFile loadTelemetriesFromDisk() {
     }
 
     operationLogger.recordSuccess();
-    return new PersistedFile(tempFile, instrumentationKey, ByteBuffer.wrap(telemetryBytes));
-  }
-
-  static boolean isInstrumentationKeyValid(String instrumentationKey) {
-    return Pattern.matches(INSTRUMENTATION_KEY_REGEX, instrumentationKey.toLowerCase());
+    return new PersistedFile(tempFile, connectionString, ByteBuffer.wrap(telemetryBytes));
   }
 
-  // reads bytes from a FileInputStream and allocates those into the buffer array byteArray.
-  private static void readFully(FileInputStream fileInputStream, byte[] byteArray, int length)
-      throws IOException {
-    if (length < 0) {
-      throw new IndexOutOfBoundsException();
+  private void deleteFile(File tempFile) {
+    if (!FileUtil.deleteFileWithRetries(tempFile)) {
+      operationLogger.recordFailure(
+          "Unable to delete file: " + tempFile.getAbsolutePath(), DISK_PERSISTENCE_LOADER_ERROR);
     }
+  }
 
-    int totalRead = 0;
-    while (totalRead < length) {
-      int numRead = fileInputStream.read(byteArray, totalRead, length - totalRead);
-      if (numRead < 0) {
-        throw new EOFException();
-      }
-
-      totalRead += numRead;
+  static boolean isConnectionStringKeyValid(String connectionString) {
+    try {
+      ConnectionString.parse(connectionString);
```

Comment:
should we change parse to return boolean? 
this doesn't indicate it was valid or not? i guess parse will throw a runtime exception when it's invalid?

Reply:
parse returns `ConnectionString` object

if it's not valid it throws an exception